### PR TITLE
Implement module configuration

### PR DESCRIPTION
**Pull request by mojtaba-nematpour**
* -- Add setting tab that redirects to module configuration
* -- Create files for module configuration
* -- Save module configuration and uninstall configs when module uninstalled
* -- Improve naming and refactor code

### CODE REVIEW

File: ps_pgzarinpal.php [hunk 1]
```diff
@@ -52,6 +52,9 @@ public function __construct()
         ];
     }
 
+    /**
+     * check module requirement and compatibility install module requirement, models, extensions and tabs
```
Comment (Arman-Hosseini) [inline ps_pgzarinpal.php]:
```suggestion
     * Installs module requirements, such as hooks, models, etc.
```

File: ps_pgzarinpal.php [hunk 2]
```diff
@@ -73,14 +76,68 @@ public function install()
         return parent::install() &&
             $this->registerHook('paymentOptions') &&
             $this->registerHook('displayPayment') &&
-            $this->registerHook('displayPaymentReturn');
+            $this->registerHook('displayPaymentReturn') &&
+            $this->installTabs();
     }
 
+    /**
+     * uninstall module models and tabs
```
Comment (Arman-Hosseini) [inline ps_pgzarinpal.php]:
```suggestion
     * Uninstall module models and tabs.
```

File: ps_pgzarinpal.php [hunk 3]
```diff
@@ -73,14 +76,68 @@ public function install()
...
+        $tabSettings->icon = 'settings';
+        $tabSettings->id_parent = $tabParent->id;
+        $tabSettings->module = $this->name;
+        $tabSettings->save();
+
+        return true;
+    }
+
+    /**
+     * uninstall module tabs
```
Comment (Arman-Hosseini) [inline ps_pgzarinpal.php]:
```suggestion
     * Uninstall module tabs.
```

**Review by Arman-Hosseini — APPROVED** [review]

**Comment by Arman-Hosseini** [comment]
Thanks @mojtaba-nematpour!

**Comment by mojtaba-nematpour** [comment]
Your Wellcome

**Pull request merged**